File: app/src/main/java/com/vkochenkov/calculator/CalculatorService.java
package com.vkochenkov.calculator;

import lombok.Getter;

@Getter
public class CalculatorService {

    private float firstNumber;
    private float secondNumber;

    private String numberStr = "0";
    private String logStr = "";

    private Operation operation;
    private boolean numberInputCompleted;

    public CalculatorService() {
        this.operation = Operation.DEFAULT;
        this.numberInputCompleted = false;
    }

    public void addSymbol(int viewId) {
        if (numberInputCompleted) {
            numberStr = "";
            numberInputCompleted = false;
        }
        if (numberStr.equals("0")) {
            numberStr = "";
        }
        if (numberStr.length() < 12) {
            switch (viewId) {
                case R.id.btnZero:
                    numberStr += "0";
                    break;
                case R.id.btnOne:
                    numberStr += "1";
                    break;
                case R.id.btnTwo:
                    numberStr += "2";
                    break;
                case R.id.btnThree:
                    numberStr += "3";
                    break;
                case R.id.btnFour:
                    numberStr += "4";
                    break;
                case R.id.btnFive:
                    numberStr += "5";
                    break;
                case R.id.btnSix:
                    numberStr += "6";
                    break;
                case R.id.btnSeven:
                    numberStr += "7";
                    break;
                case R.id.btnEight:
                    numberStr += "8";
                    break;
                case R.id.btnNine:
                    numberStr += "9";
                    break;
                case R.id.btnPoint:
                    if (numberStr.equals("")) {
                        numberStr = "0";
                    }
                    if (!numberStr.contains(".")) {
                        numberStr += ".";
                    }
                    break;
            }
        }
    }

    public void delSymbols(int viewId) {
        switch (viewId) {
            case R.id.btnDel:
                if (numberStr.length() > 1) {
                    numberStr = numberStr.substring(0, numberStr.length() - 1);
                } else {
                    numberStr = "0";
                }
                break;
            case R.id.btnClear:
                numberStr = "0";
                logStr = "";
                break;
        }
    }

    public void selectOperation(int viewId) {
        if (!numberStr.equals("")) {
            switch (viewId) {
                case R.id.btnChangeSign:
                    //todo
                    break;
                case R.id.btnPlus:
                    firstNumber = Float.parseFloat(numberStr);
                    operation = Operation.ADDITION;
                    logStr = firstNumber + operation.getSymbol();
                    break;
                case R.id.btnMinus:
                    firstNumber = Float.parseFloat(numberStr);
                    operation = Operation.SUBTRACTION;
                    logStr = firstNumber + operation.getSymbol();
                    break;
                case R.id.btnMultiply:
                    firstNumber = Float.parseFloat(numberStr);
                    operation = Operation.MULTIPLICATION;
                    logStr = firstNumber + operation.getSymbol();
                    break;
                case R.id.btnDivision:
                    firstNumber = Float.parseFloat(numberStr);
                    operation = Operation.DIVISION;
                    logStr = firstNumber + operation.getSymbol();
                    break;
                case R.id.btnEqual:
                    secondNumber = Float.parseFloat(numberStr);
                    calculate(operation);
                    if (operation!=Operation.EQUALS) {
                        logStr = firstNumber + operation.getSymbol() + secondNumber + Operation.EQUALS.getSymbol() + numberStr;
                    }
                    operation = Operation.EQUALS;
                    break;
            }
            numberInputCompleted = true;
        } else {
            logStr = "Something went wrong!";
        }

    }

    private void calculate(Operation operation) {
        float answer = secondNumber;
        switch (operation) {
            case ADDITION:
                answer = firstNumber + secondNumber;
                break;
            case SUBTRACTION:
                answer = firstNumber - secondNumber;
                break;
            case MULTIPLICATION:
                answer = firstNumber * secondNumber;
                break;
            case DIVISION:
                answer = firstNumber / secondNumber;
                break;
        }
        numberStr = Float.toString(answer);

    }
}

File: app/src/main/java/com/vkochenkov/calculator/CalculatorService.kt
package com.vkochenkov.calculator

class CalculatorService(var numberStr: String = "0", var logStr: String = "") {
    private var firstNumber = 0f
    private var secondNumber = 0f
    private var operation: Operation = Operation.DEFAULT
    private var numberInputCompleted: Boolean = false

    fun addSymbol(viewId: Int) {
        if (numberInputCompleted) {
            numberStr = ""
            numberInputCompleted = false
        }
        if (numberStr == "0") {
            numberStr = ""
        }
        if (numberStr.length < 12) {
            when (viewId) {
                R.id.btnZero -> numberStr += "0"
                R.id.btnOne -> numberStr += "1"
                R.id.btnTwo -> numberStr += "2"
                R.id.btnThree -> numberStr += "3"
                R.id.btnFour -> numberStr += "4"
                R.id.btnFive -> numberStr += "5"
                R.id.btnSix -> numberStr += "6"
                R.id.btnSeven -> numberStr += "7"
                R.id.btnEight -> numberStr += "8"
                R.id.btnNine -> numberStr += "9"
                R.id.btnPoint -> {
                    if (numberStr == "") {
                        numberStr = "0"
                    }
                    if (!numberStr.contains(".")) {
                        numberStr += "."
                    }
                }
            }
        }
    }

    fun delSymbols(viewId: Int) {
        when (viewId) {
            R.id.btnDel -> {
                if (numberStr.contains("Infinity").not() and numberStr.contains("NaN").not()) {
                    numberStr =
                            if (numberStr.length > 1) {
                                numberStr.substring(0, numberStr.length - 1)
                            } else {
                                "0"
                            }
                }
            }
            R.id.btnClear -> {
                numberStr = "0"
                logStr = ""
            }
        }
    }

    fun selectOperation(viewId: Int) {
        if (numberStr != "") {
            when (viewId) {
                R.id.btnChangeSign -> {
                    //todo
                }
                R.id.btnPlus -> {
                    firstNumber = tryNumberStringToFloat()
                    operation = Operation.ADDITION
                    logStr = firstNumber.toString() + operation.symbol
                }
                R.id.btnMinus -> {
                    firstNumber = tryNumberStringToFloat()
                    operation = Operation.SUBTRACTION
                    logStr = firstNumber.toString() + operation.symbol
                }
                R.id.btnMultiply -> {
                    firstNumber = tryNumberStringToFloat()
                    operation = Operation.MULTIPLICATION
                    logStr = firstNumber.toString() + operation.symbol
                }
                R.id.btnDivision -> {
                    firstNumber = tryNumberStringToFloat()
                    operation = Operation.DIVISION
                    logStr = firstNumber.toString() + operation.symbol
                }
                R.id.btnEqual -> {
                    secondNumber = tryNumberStringToFloat()
                    calculate(operation)
                    if (operation !== Operation.EQUALS) {
                        logStr = firstNumber.toString() + operation.symbol + secondNumber + Operation.EQUALS.symbol + numberStr
                    }
                    operation = Operation.EQUALS
                }
            }
            numberInputCompleted = true
        } else {
            logStr = "Something went wrong!"
        }
    }

    private fun calculate(operation: Operation) {
        var answer = secondNumber
        when (operation) {
            Operation.ADDITION -> answer = firstNumber + secondNumber
            Operation.SUBTRACTION -> answer = firstNumber - secondNumber
            Operation.MULTIPLICATION -> answer = firstNumber * secondNumber
            Operation.DIVISION -> answer = firstNumber / secondNumber
        }
        numberStr = java.lang.Float.toString(answer)
    }

    private fun tryNumberStringToFloat(): Float {
        var targetField: Float = 0f
        try {
            targetField = numberStr.toFloat()
        } catch (e: NumberFormatException) {
            logStr = "Something went wrong!"
        }
        return targetField
    }
}